Error: [$injector:strictdi] Socket is not using explicit annotation and cannot be invoked in strict mode

- [x] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 4.0.4
Node | 6.3.1
npm | 3.10.3
Operating System | Ubuntu 14.04

Item | Answer
----- | -----
Transpiler | Babel
Markup | HTML
CSS | CSS
Router | ngRoute
Client Tests | Jasmine
DB | MongoDB
Auth | Y



When I run `grunt serve`,  the page is loaded without any problem, but I am getting this error on Chrome's console: 

```
angular.js:13920 Error: [$injector:strictdi] Socket is not using explicit annotation and cannot be invoked in strict mode
http://errors.angularjs.org/1.5.8/$injector/strictdi?p0=Socket
    at http://localhost:3000/vendor.bundle.js:195:13
    at Function.annotate [as $$annotate] (http://localhost:3000/vendor.bundle.js:4079:18)
    at injectionArgs (http://localhost:3000/vendor.bundle.js:4806:37)
    at Object.invoke (http://localhost:3000/vendor.bundle.js:4837:19)
    at Object.enforcedReturnValue [as $get] (http://localhost:3000/vendor.bundle.js:4684:38)
    at Object.invoke (http://localhost:3000/vendor.bundle.js:4845:20)
    at http://localhost:3000/vendor.bundle.js:4644:38
    at getService (http://localhost:3000/vendor.bundle.js:4791:40)
    at injectionArgs (http://localhost:3000/vendor.bundle.js:4815:59)
    at Object.invoke (http://localhost:3000/vendor.bundle.js:4837:19)
    at $controllerInit (http://localhost:3000/vendor.bundle.js:10481:35)
    at nodeLinkFn (http://localhost:3000/vendor.bundle.js:9390:35)
    at compositeLinkFn (http://localhost:3000/vendor.bundle.js:8747:14)
    at publicLinkFn (http://localhost:3000/vendor.bundle.js:8627:31)
    at Object.link (http://localhost:3000/vendor.bundle.js:42606:8)
    at http://localhost:3000/vendor.bundle.js:1374:19
    at invokeLinkFn (http://localhost:3000/vendor.bundle.js:10061:10)
    at nodeLinkFn (http://localhost:3000/vendor.bundle.js:9462:12)
    at compositeLinkFn (http://localhost:3000/vendor.bundle.js:8747:14)
    at publicLinkFn (http://localhost:3000/vendor.bundle.js:8627:31)
    at lazyCompilation (http://localhost:3000/vendor.bundle.js:8971:26)
    at boundTranscludeFn (http://localhost:3000/vendor.bundle.js:8764:17)
    at controllersBoundTransclude (http://localhost:3000/vendor.bundle.js:9512:21)
    at update (http://localhost:3000/vendor.bundle.js:42555:26)
    at Scope.$broadcast (http://localhost:3000/vendor.bundle.js:18132:29)
    at http://localhost:3000/vendor.bundle.js:42195:27
    at processQueue (http://localhost:3000/vendor.bundle.js:16510:29)
    at http://localhost:3000/vendor.bundle.js:16526:28
    at Scope.$eval (http://localhost:3000/vendor.bundle.js:17809:29)
    at Scope.$digest (http://localhost:3000/vendor.bundle.js:17622:32)
    at Scope.$apply (http://localhost:3000/vendor.bundle.js:17917:25)
    at bootstrapApply (http://localhost:3000/vendor.bundle.js:1888:16)
    at Object.invoke (http://localhost:3000/vendor.bundle.js:4845:20)
    at doBootstrap (http://localhost:3000/vendor.bundle.js:1886:15)
    at Object.bootstrap (http://localhost:3000/vendor.bundle.js:1906:13)
    at http://localhost:3000/app.bundle.js:86:22
    at HTMLDocument.trigger (http://localhost:3000/vendor.bundle.js:3334:8)
    at defaultHandlerWrapper (http://localhost:3000/vendor.bundle.js:3624:12)
    at HTMLDocument.eventHandler (http://localhost:3000/vendor.bundle.js:3612:10) <div ng-view="" class="ng-scope">
```

And a list of warning on my terminal:

```
/home/angular-fullstack/server/app.js
  16:17  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/index.js
  4:11  warning  Unexpected use of process.env  no-process-env
  4:34  warning  Unexpected use of process.env  no-process-env

/home/angular-fullstack/server/routes.js
  24:33  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/auth/auth.service.js
  22:37  warning  Unexpected string concatenation  prefer-template
  26:37  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/config/express.js
  39:20  warning  Unexpected string concatenation  prefer-template
  67:25  warning  Unexpected use of process.env    no-process-env
  97:14  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/api/thing/thing.controller.js
   29:7  warning  Avoid using Function.prototype.apply, instead use Reflect.apply     prefer-reflect
   91:5  warning  Avoid using the delete keyword, instead use Reflect.deleteProperty  prefer-reflect
  102:5  warning  Avoid using the delete keyword, instead use Reflect.deleteProperty  prefer-reflect

/home/angular-fullstack/server/api/thing/thing.events.js
  28:22  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/api/user/user.events.js
  28:21  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/server/api/user/user.model.js
  162:15  warning  Use the rest parameters instead of 'arguments'  prefer-rest-params
  163:18  warning  Use the rest parameters instead of 'arguments'  prefer-rest-params
  165:22  warning  Use the rest parameters instead of 'arguments'  prefer-rest-params
  166:18  warning  Use the rest parameters instead of 'arguments'  prefer-rest-params
  206:14  warning  Unexpected sync method: 'pbkdf2Sync'            no-sync

✖ 19 problems (0 errors, 19 warnings)

[13:39:07] Finished 'lint:scripts:server' after 1.91 s
[13:39:07] 
/home/angular-fullstack/client/app/main/main.component.js
  35:23  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/client/components/auth/auth.service.js
  192:14  warning  Avoid using Function.prototype.apply, instead use Reflect.apply  prefer-reflect
  192:14  warning  use the spread operator instead of the '.apply()'                prefer-spread
  192:39  warning  Avoid using Function.prototype.apply, instead use Reflect.apply  prefer-reflect
  192:66  warning  Use the rest parameters instead of 'arguments'                   prefer-rest-params
  201:14  warning  Unexpected sync method: 'hasRoleSync'                            no-sync

/home/angular-fullstack/client/components/auth/interceptor.service.js
  11:40  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/client/components/modal/modal.service.js
  42:22  warning  Avoid using Function.prototype.call, instead use Reflect.apply   prefer-reflect
  42:49  warning  Use the rest parameters instead of 'arguments'                   prefer-rest-params
  50:21  warning  Unexpected string concatenation                                  prefer-template
  68:13  warning  Avoid using Function.prototype.apply, instead use Reflect.apply  prefer-reflect

/home/angular-fullstack/client/components/socket/socket.service.js
  38:17  warning  Unexpected string concatenation  prefer-template
  60:17  warning  Unexpected string concatenation  prefer-template
  75:33  warning  Unexpected string concatenation  prefer-template
  76:33  warning  Unexpected string concatenation  prefer-template

/home/angular-fullstack/client/components/util/util.service.js
  59:99  warning  '===' should be placed at the beginning of the line  operator-linebreak

✖ 16 problems (0 errors, 16 warnings)
```

Can I get some help here please?

Thanks.

--- a/templates/app/client/components/socket(socketio)/socket.service.js
+++ b/templates/app/client/components/socket(socketio)/socket.service.js
@@ -4,6 +4,7 @@ import angular from 'angular';
 import io from 'socket.io-client';
 
 function Socket(socketFactory) {
+    'ngInject';
     // socket.io now auto-configures its connection when we ommit a connection url
     var ioSocket = io('', {
       // Send auth token on connection, you will need to DI the Auth service above

